nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,503 @@
+    boot.kernel.sysctl."net.core.netdev_max_backlog" = mkDefault 100000;
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.videobridge.openFirewall
+      [ jvbConfig.videobridge.ice.tcp.port ];
+    networking.firewall.allowedUDPPorts = mkIf cfg.videobridge.openFirewall
+      [ jvbConfig.videobridge.ice.udp.port ];
+
+    systemd.services.jicofo = let
+      jicofoProps = {
+        "-Dnet.java.sip.communicator.SC_HOME_DIR_LOCATION" = "/etc/jitsi";
+        "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "jicofo";
+        "-Djava.util.logging.config.file" = "/etc/jitsi/jicofo/logging.properties";
+        "-Djavax.net.ssl.trustStore" = "/var/lib/jitsi-meet/prosody-ca.jks";
+        "-Djavax.net.ssl.trustStorePassword" = "javapls";
+      };
+      jicofoArgs = {
+        "--host" = "localhost";
+        "--domain" = cfg.hostName;
+        "--secret" = "\${JICOFO_COMPONENT_SECRET}";
+        "--user_domain" = "auth.${cfg.hostName}";
+        "--user_name" = "focus";
+        "--user_password" = "\${JICOFO_USER_SECRET}";
+      };
+    in
+    {
+      description = "JItsi COnference FOcus";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      restartTriggers = [
+        config.environment.etc."jitsi/jicofo/sip-communicator.properties".source
+      ];
+      environment.JAVA_SYS_PROPS = attrsToArgs jicofoProps;
+
+      serviceConfig = {
+        Type = "exec";
+        ExecStart = "${pkgs.jicofo}/bin/jicofo ${attrsToArgs jicofoArgs}";
+
+        DynamicUser = true;
+        User = "jicofo";
+        Group = "jitsi-meet";
+
+        EnvironmentFile = [ "/var/lib/jitsi-meet/secrets" ];
+
+        CapabilityBoundingSet = "";
+        NoNewPrivileges = true;
+        ProtectSystem = "strict";
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        ProtectHostname = true;
+        ProtectKernelTunables = true;
+        ProtectKernelModules = true;
+        ProtectControlGroups = true;
+        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" "AF_UNIX" ];
+        RestrictNamespaces = true;
+        LockPersonality = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+      };
+    };
+
+    environment.etc."jitsi/jicofo/sip-communicator.properties".source =
+      pkgs.writeText "sip-communicator.properties" (
+        concatStringsSep "\n" (mapAttrsToList (k: v: "${k}=${v}") cfg.jicofo.config)
+      );
+    environment.etc."jitsi/jicofo/logging.properties".source =
+      mkDefault "${pkgs.jicofo}/etc/jitsi/jicofo/logging.properties-journal";
+
+    assertions = [{
+      message = "Strings that start with __ENV__ must also end with __ENV__";
+      assertion = let
+        p = x: if isString x then (hasPrefix "__ENV__" x) == (hasSuffix "__ENV__" x)
+          else if isList x then builtins.all p x
+          else if isAttrs x then p (builtins.attrValues x)
+          else true;
+      in p jvbConfig;
+    }];
+  };
```

Comment:
As expessed above, if we'd use HOCON's environment variable syntax from the start, we wouldn't need to add our own assertions here.

Reply:
Seems we need to keep that specific environment variable substitution as explained in https://github.com/NixOS/nixpkgs/pull/82920#discussion_r411491297, this might probably still need a comment.

Reply:
Rewrote the HOCON handling so assertion is no longer needed to generate valid HOCON/JSON.